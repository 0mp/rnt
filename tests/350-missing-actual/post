. ../post.functions

trap "rm -f test/*.actual test/*.diff" EXIT

assert_file test/out.expected
assert_file test/out.actual
assert_file test/out.diff
assert_file test/out.diff.tmp

assert_file test/err.expected
assert_file test/err.actual
assert_file test/err.diff
assert_file test/err.diff.tmp

assert_file test/exit.expected
assert_file test/exit.actual
assert_file test/exit.diff
assert_file test/exit.diff.tmp

assert_file test/extra.expected
assert_file test/extra.actual
assert_file test/extra.diff
assert_file test/extra.diff.tmp
assert_same expected/extra.diff test/extra.diff
