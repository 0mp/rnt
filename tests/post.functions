cleanup()
{
  rm -f "$PWD/test/exit.actual"
  rm -f "$PWD/test/err.actual"
  rm -f "$PWD/test/out.actual"
  rm -f "$PWD/test/exit.diff"
  rm -f "$PWD/test/err.diff"
  rm -f "$PWD/test/out.diff"
}

assert_file()
{
  if test -e "${1:?}"; then
    echo "present: $1"
  else
    echo "missing: $1"
  fi
}

assert_nonexistent()
{
  assert_file ${1+"$@"}
}

assert_same()
{
  diff -u "${1:?}" "${2:?}"
}

