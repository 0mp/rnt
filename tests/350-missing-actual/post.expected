present: test/out.expected
present: test/out.actual
missing: test/out.diff
missing: test/out.diff.tmp
present: test/err.expected
present: test/err.actual
missing: test/err.diff
missing: test/err.diff.tmp
present: test/exit.expected
present: test/exit.actual
missing: test/exit.diff
missing: test/exit.diff.tmp
present: test/extra.expected
missing: test/extra.actual
present: test/extra.diff
missing: test/extra.diff.tmp
